src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/ContextHolder.cs

```
@@ -42,7 +45,63 @@ public unsafe byte[] GetBytes()
         Span<byte> byteSpan = MemoryMarshal.AsBytes(structSpan);
         return byteSpan.ToArray();
     }
-    public IPlatformAgnosticContext Clone() => new CotnextHolder<T>() { Context = Context };
+    public IPlatformAgnosticContext Clone() => new ContextHolder<T>() { Context = Context };
     public void Clear() => Context = default;
     public void Unwind(Target target) => Context.Unwind(target);
+
+    public bool TrySetRegister(Target target, string fieldName, TargetNUInt value)
+    {
+        if (typeof(T).GetField(fieldName) is not FieldInfo field) return false;
+        switch (field.FieldType)
+        {
+            case Type t when t == typeof(ulong) && target.PointerSize == sizeof(ulong):
+                field.SetValueDirect(__makeref(Context), value.Value);
+                return true;
+            case Type t when t == typeof(uint) && target.PointerSize == sizeof(uint):
+                field.SetValueDirect(__makeref(Context), (uint)value.Value);
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    public bool TryReadRegister(Target target, string fieldName, out TargetNUInt value)
+    {
+        value = default;
+        if (typeof(T).GetField(fieldName) is not FieldInfo field) return false;
+        if (field.GetValue(Context) is not object fieldValue) return false;
+        if (fieldValue is ulong ul && target.PointerSize == sizeof(ulong))
+        {
+            value = new(ul);
+            return true;
+        }
+        if (fieldValue is uint ui && target.PointerSize == sizeof(uint))
+        {
+            value = new(ui);
+            return true;
+        }
+        return false;
+    }
+
+    public bool Equals(ContextHolder<T>? other)
+    {
+        if (other is null)
+        {
+            return false;
+        }
+
```

If you want, you can eliminate this whole check by changing the `if` below to `if (GetType() != other?.GetType())` - a `null` `other` will have a `null` `other?.GetType()` which can't ever match our own `GetType()`.

Change not necessary though.

Is there a reason why this `Equals` implementation is written in this specific way (exact type match only) even though `ContextHolder` isn't `sealed`? It's a little weird to see it, and I'd prefer to block this class of failure by sealing the class if we can do that because then the attempt to do the wrong thing will fail robustly at compile time instead of silently failing `Equals` checks.

---

I'll change this class to `sealed`